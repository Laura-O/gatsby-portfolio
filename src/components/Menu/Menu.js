import React from 'react';
import Link from 'gatsby-link';
import styled from 'styled-components';
import Fullscreen from '../Fullscreen/Fullscreen';
import * as constants from '../../layouts/shared/style-constants';

const Item = styled.div`
  color: white;
  position: relative;
  font-size: 3rem;
  line-height: 2;
  transition: background-size 0.4s ease-in;
  background: linear-gradient(
      to bottom,
      transparent 50%,
      ${constants.linkColor} 0
    )
    center center/0% 75% no-repeat;
  padding: 0 5px 2px 5px;
  cursor: pointer;
  letter-spacing: 0.08rem;

  &:hover {
    background-size: 100% 50%;
  }
  &:active {
    background-size: 80% 100%;
  }

  a {
    text-decoration: none;

    &:hover {
      color: white;
    }
  }
`;

export default function Menu(props) {
  const linkProps = {
    onClick: props.toggleNav,
  };

  return (
    <Fullscreen active={props.active}>
      <nav>
        <Item>
          <Link {...linkProps} to="/">
            Home
          </Link>
        </Item>
        <Item>
          <Link {...linkProps} to="/about">
            About
          </Link>
        </Item>
        <Item>
          <Link {...linkProps} to="/blog">
            Blog
          </Link>
        </Item>
        <Item>
          <Link {...linkProps} to="/projects">
            Projects
          </Link>
        </Item>
        <Item>
          <Link {...linkProps} to="/contact">
            Contact
          </Link>
        </Item>
      </nav>
    </Fullscreen>
  );
}
