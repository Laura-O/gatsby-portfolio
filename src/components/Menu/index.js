import React from 'react'
import Link from 'gatsby-link'

import Fullscreen from '../Fullscreen'

import styles from './styles.module.scss'

const Menu = ({ active, toggleNav }) => {
  const linkProps = {
    onClick: toggleNav,
  }

  return (
    <Fullscreen className={active ? styles.menu : styles.disable}>
      <nav className={styles.nav}>
        <div className={styles.item}>
          <Link {...linkProps} to="/">
            Home
          </Link>
        </div>
        <div className={styles.item}>
          <Link {...linkProps} to="/about">
            About
          </Link>
        </div>
        <div className={styles.item}>
          <Link {...linkProps} to="/projects">
            Projects
          </Link>
        </div>
        <div className={styles.item}>
          <Link {...linkProps} to="/contact">
            Contact
          </Link>
        </div>
      </nav>
    </Fullscreen>
  )
}

export default Menu
