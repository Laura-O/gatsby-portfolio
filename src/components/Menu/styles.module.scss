@import '../../scss/variables.scss';

.menu {
  background: $colorBlack;
  color: $colorWhite;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 3;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.disable {
  display: none;
}

.item {
  color: white;
  position: relative;
  font-size: 3rem;
  line-height: 2;
  transition: background-size 0.4s ease-in;
  background: linear-gradient(to bottom, transparent 50%, $linkColor 0) center
    center/0% 75% no-repeat;
  padding: 0 5px 2px 5px;
  cursor: pointer;
  letter-spacing: 0.08rem;

  &:hover {
    background-size: 100% 50%;
  }
  &:active {
    background-size: 80% 100%;
  }

  a {
    text-decoration: none;
  }
}

.active {
  border-bottom: 4px solid $colorWhite;
}

@media (--small) {
  .item {
    font-size: 2.2857rem;
  }
  .active {
    border-width: 2px;
  }
}
