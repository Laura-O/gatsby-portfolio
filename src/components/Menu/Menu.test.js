import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

test('rendering', () => {
  const wrapper = shallow(<Menu active toggleNav={jest.fn()} />);
  expect(wrapper).toMatchSnapshot();
});
