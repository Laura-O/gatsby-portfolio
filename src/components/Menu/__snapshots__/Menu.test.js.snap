// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu
    active={true}
    toggleNav={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": true,
      "children": <nav>
        <styled.div>
          <UNDEFINED
            onClick={[MockFunction]}
            to="/about"
          >
            About
          </UNDEFINED>
        </styled.div>
        <styled.div>
          <UNDEFINED
            onClick={[MockFunction]}
            to="/blog"
          >
            Blog
          </UNDEFINED>
        </styled.div>
        <styled.div>
          <UNDEFINED
            onClick={[MockFunction]}
            to="/projects"
          >
            Projects
          </UNDEFINED>
        </styled.div>
        <styled.div>
          <UNDEFINED
            onClick={[MockFunction]}
            to="/galleries"
          >
            Galleries
          </UNDEFINED>
        </styled.div>
      </nav>,
      "data-testid": "fullscreen",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.div>
            <UNDEFINED
              onClick={[MockFunction]}
              to="/about"
            >
              About
            </UNDEFINED>
          </styled.div>,
          <styled.div>
            <UNDEFINED
              onClick={[MockFunction]}
              to="/blog"
            >
              Blog
            </UNDEFINED>
          </styled.div>,
          <styled.div>
            <UNDEFINED
              onClick={[MockFunction]}
              to="/projects"
            >
              Projects
            </UNDEFINED>
          </styled.div>,
          <styled.div>
            <UNDEFINED
              onClick={[MockFunction]}
              to="/galleries"
            >
              Galleries
            </UNDEFINED>
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <UNDEFINED
              onClick={[MockFunction]}
              to="/about"
            >
              About
            </UNDEFINED>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "About",
              "onClick": [MockFunction],
              "to": "/about",
            },
            "ref": null,
            "rendered": "About",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <UNDEFINED
              onClick={[MockFunction]}
              to="/blog"
            >
              Blog
            </UNDEFINED>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Blog",
              "onClick": [MockFunction],
              "to": "/blog",
            },
            "ref": null,
            "rendered": "Blog",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <UNDEFINED
              onClick={[MockFunction]}
              to="/projects"
            >
              Projects
            </UNDEFINED>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Projects",
              "onClick": [MockFunction],
              "to": "/projects",
            },
            "ref": null,
            "rendered": "Projects",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <UNDEFINED
              onClick={[MockFunction]}
              to="/galleries"
            >
              Galleries
            </UNDEFINED>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Galleries",
              "onClick": [MockFunction],
              "to": "/galleries",
            },
            "ref": null,
            "rendered": "Galleries",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          },
          "type": [Function],
        },
      ],
      "type": "nav",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": true,
        "children": <nav>
          <styled.div>
            <UNDEFINED
              onClick={[MockFunction]}
              to="/about"
            >
              About
            </UNDEFINED>
          </styled.div>
          <styled.div>
            <UNDEFINED
              onClick={[MockFunction]}
              to="/blog"
            >
              Blog
            </UNDEFINED>
          </styled.div>
          <styled.div>
            <UNDEFINED
              onClick={[MockFunction]}
              to="/projects"
            >
              Projects
            </UNDEFINED>
          </styled.div>
          <styled.div>
            <UNDEFINED
              onClick={[MockFunction]}
              to="/galleries"
            >
              Galleries
            </UNDEFINED>
          </styled.div>
        </nav>,
        "data-testid": "fullscreen",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <styled.div>
              <UNDEFINED
                onClick={[MockFunction]}
                to="/about"
              >
                About
              </UNDEFINED>
            </styled.div>,
            <styled.div>
              <UNDEFINED
                onClick={[MockFunction]}
                to="/blog"
              >
                Blog
              </UNDEFINED>
            </styled.div>,
            <styled.div>
              <UNDEFINED
                onClick={[MockFunction]}
                to="/projects"
              >
                Projects
              </UNDEFINED>
            </styled.div>,
            <styled.div>
              <UNDEFINED
                onClick={[MockFunction]}
                to="/galleries"
              >
                Galleries
              </UNDEFINED>
            </styled.div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <UNDEFINED
                onClick={[MockFunction]}
                to="/about"
              >
                About
              </UNDEFINED>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "About",
                "onClick": [MockFunction],
                "to": "/about",
              },
              "ref": null,
              "rendered": "About",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <UNDEFINED
                onClick={[MockFunction]}
                to="/blog"
              >
                Blog
              </UNDEFINED>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Blog",
                "onClick": [MockFunction],
                "to": "/blog",
              },
              "ref": null,
              "rendered": "Blog",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <UNDEFINED
                onClick={[MockFunction]}
                to="/projects"
              >
                Projects
              </UNDEFINED>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Projects",
                "onClick": [MockFunction],
                "to": "/projects",
              },
              "ref": null,
              "rendered": "Projects",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <UNDEFINED
                onClick={[MockFunction]}
                to="/galleries"
              >
                Galleries
              </UNDEFINED>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Galleries",
                "onClick": [MockFunction],
                "to": "/galleries",
              },
              "ref": null,
              "rendered": "Galleries",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "type": [Function],
          },
        ],
        "type": "nav",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
